src/AbstractPluginManager.php

```diff
@@ -110,16 +116,18 @@ public function __construct($configInstanceOrParentLocator = null, array $config
     /**
      * Override configure() to validate service instances.
      *
-     * If an instance passed in the `services` configuration is invalid for the
-     * plugin manager, this method will raise an InvalidServiceException.
-     *
-     * {@inheritDoc}
-     *
-     * @throws InvalidServiceException
+     * @param  array $config
+     * @psalm-param ServiceManagerConfiguration $config
+     * @return self
+     * @throws InvalidServiceException If an instance passed in the `services` configuration is invalid for the
+     *                                 plugin manager.
+     * @throws ContainerModificationsNotAllowedException If the allow override flag has been toggled off, and a
+     *                                                   service instanceexists for a given service.
      */
     public function configure(array $config)
     {
         if (isset($config['services'])) {
+            /** @psalm-suppress MixedAssignment */
```

Comment:
This should be fixable: `@psalm-param ServiceManagerConfiguration|array{services: ServiceManagerConfiguration} $config`

Reply:
That would be a false type as `services` does not contain another `ServiceManagerConfiguration`. 
It contains `object|array` which is properly annotated but somehow psalm refuses to infer it from the parameter type.

Reply:
Ah, I misread it - thought it was a different nesting level